### PR TITLE
Enable better roundtripping of CSC2 and SQL data type names.

### PR DESCRIPTION
**Pull request by mistachkin**
Signed-off-by: mistachkin <joe@mistachkin.com>

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/66eee37/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>7/380 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>yast<br>upsert<br>ddl_no_csc2<br>ddl_csc2<br>comdb2sys_queueodh_generated<br>comdb2sys<br>cdb2api</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/66eee37/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/66eee37/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2556'>Resubmit</a></p>

File: sqlite/src/comdb2build.c [hunk 1]
```diff
@@ -2724,16 +2724,18 @@ enum {
...
+    XMACRO_TYPE(SQL_TYPE_SMALLFLOAT, "smallfloat", "float", 0)                 \
     /* Additional types mapped to a Comdb2 type. */                            \
-    XMACRO_TYPE(SQL_TYPE_VARCHAR, "varchar", "cstring",                        \
-                FLAG_ALLOW_ARRAY | FLAG_QUOTE_DEFAULT | FLAG_EXTRA_BYTE)       \
     XMACRO_TYPE(SQL_TYPE_CHAR, "char", "cstring",                              \
                 FLAG_ALLOW_ARRAY | FLAG_QUOTE_DEFAULT | FLAG_EXTRA_BYTE)       \
     XMACRO_TYPE(SQL_TYPE_TEXT, "text", "vutf8",                                \
                 FLAG_ALLOW_ARRAY | FLAG_QUOTE_DEFAULT)                         \
+    XMACRO_TYPE(SQL_TYPE_VARCHAR, "varchar", "vutf8",                          \
+                FLAG_ALLOW_ARRAY | FLAG_QUOTE_DEFAULT)                         \
```
Comment (nirbhayc) [inline sqlite/src/comdb2build.c:2734]: So, we have:
```
sql(char) -> csc2(cstring)
sql(varchar) -> csc2(vutf8)
sql(text) -> csc2(vutf8)
```
I think that's ok.

File: tests/cdb2_coltype.test/test.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
```
Comment (nirbhayc) [inline tests/cdb2_coltype.test/test.sh:1]: Could you use tests/tools/compare_results.sh instead?

File: tests/ddl_csc2.test/t01_create_table.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS t4;
+DROP TABLE IF EXISTS t5;
+DROP TABLE IF EXISTS t6;
+
+CREATE TABLE t4(x1 SMALLINT, x2 BIGINT, x3 INT, x4 LARGEINT);$$
```
Comment (nirbhayc) [inline tests/ddl_csc2.test/t01_create_table.sql:5]: So, both bigint and largeint map to largeint. This is intentional. Right?

File: tools/cdb2sql/cdb2sql.cpp [hunk 4]
```diff
@@ -943,31 +966,52 @@ static void print_column(FILE *f, cdb2_hndl_tp *hndl, int col)
         return;
     }
 
-    switch (cdb2_column_type(hndl, col)) {
+    int type = cdb2_column_type(hndl, col);
+    std::string typeStr;
+
+    if (printmode & DISP_COLTYPE) {
+        if (type >= MIN_COL_TYPE_NAME && type <= MAX_COL_TYPE_NAME) {
+            typeStr = col_type_names[type];
```
Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.cpp:974]: `col_type_names[type] + " "` should do as well.

**Review by nirbhayc — APPROVED** [review]
Few minor suggestions for you to consider. Looks good otherwise. Approved.

File: tests/cdb2_coltype.test/test.sh (hunk 2 above)
Comment (mistachkin) [reply tests/cdb2_coltype.test/test.sh:1]: Yes, I'll try to do that.

File: tests/ddl_csc2.test/t01_create_table.sql (hunk 3 above)
Comment (mistachkin) [reply tests/ddl_csc2.test/t01_create_table.sql:4]: yes

File: sqlite/src/comdb2build.c (hunk 1 above)
Comment (mistachkin) [reply sqlite/src/comdb2build.c]: Removed this change as "out of scope"

File: tools/cdb2sql/cdb2sql.cpp (hunk 4 above)
Comment (mistachkin) [reply tools/cdb2sql/cdb2sql.cpp]: Just tried that, it does not appear to work (it won't compile).

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/80aca71/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/380 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>ddl_csc2<br>ddl_no_csc2<br>comdb2sys<br>comdb2sys_queueodh_generated</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/80aca71/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/80aca71/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2556'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2d5a4e4/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2d5a4e4/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2d5a4e4/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2556'>Resubmit</a></p>

**Pull request merged**